spec/controllers/events_controller_spec.rb

```diff
@@ -2,16 +2,35 @@
 
 RSpec.describe EventsController, type: :controller do
   describe "GET #index" do
-    let(:upcoming_events) { double }
+    context "keyword is empty" do
+      let(:upcoming_events) { double }
 
-    before do
-      expect(Event).to receive(:upcoming).and_return upcoming_events
-      get :index
+      before do
+        expect(Event).to receive(:upcoming).and_return upcoming_events
+        get :index
+      end
+
+      it "is successful and assigns upcoming events" do
+        expect(response).to be_success
+        expect(assigns(:events)).to eq upcoming_events
+        expect(assigns(:keyword)).to be_nil
+      end
     end
 
-    it "is successful and assigns upcoming events" do
-      expect(response).to be_success
-      expect(assigns(:events)).to eq upcoming_events
+    context "keyword is not empty" do
```

Comment:
Don't forget to test `Event.search()` itself too. And make sure that you consider edge cases such as capitalization, events that match one of keywords but not the other, events that expire etc.